### ISSUE TITLE
Disambiguate quickSuggestion from editor.action.triggerSuggest

### ISSUE THREAD

**Issue by blois** [microsoft/vscode#86734]
From a CompletionItemProvider it would be useful to be able to disambiguate completions triggered by quickSuggest and manual invocation of the command (via ctrl-shift). This could be used if specific completions are expensive or particularly low confidence.

Currently quickSuggest suggestions and ctrl-space suggestions both come through with CompletionContext.triggerKind = 0 and triggerCharacter = undefined.

**Comment by jrieken** [comment]
Yes - we never saw a need for differentiating between those use-cases whereas the others have concrete ones. So, why do you need to know this?

**Comment by blois** [comment]
The concrete scenario is that we have completions on code which contains nested languages. The completion logic for the nested language may not be as high quality as the surrounding code so we'd like to only provide completions if the user explicitly requested while inside of the nested context.

An example is a system which provides completions for SQL strings within Javascript such as:
https://www.w3schools.com/nodejs/nodejs_mysql_select.asp

**Comment by jrieken** [comment]
That's an interesting idea but needs some thinking. We have a related request with completions from a fast and slow source, eg. file completions vs project completions, and the same approach good be used there. However, I am afraid that users are confused by such a subtle difference...
